experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -552,6 +552,24 @@ export function applyTypesFromContext(
 			type,
 			fields: new Map(children.length > 0 ? [[primary.key, children]] : []),
 		};
+	} else if (data instanceof Map) {
```

Comment:
@CraigMacomber hoping you can provide feedback / guidance here. Noah and I thought this was likely the best place to handle JS Map inputs. The non-proxy API won't allow such inputs per the type-system, but if we ever allowed that, it should presumably be fine? And for the proxy API, this enables Map inputs without having to re-implement the rest of the recursive logic here.

Reply:
this entire file is logic that should be superseded by Noah's factory stuff. Also this is just intended for making nodes out of object literals with strong types from  SchemaAware's flexiable mode. Adding support for maps here doesn't seem very useful (they are already supported as object literals, which is the intended use of this API) and programmatic non schema aware creators of trees shouldn't use this contextually typed format (they should make something else, like MapTree, then wrap it in a cursor).

If you do want to add this here (and I don't think its worth doing) you would also want to update the schema aware APi we use with it so users (of the old editiable tree API and initlTree until we change it) could actually benefit from your feature.

Reply:
Per our offline conversation, we've agreed that this isn't the correct long-term home for this logic, but will suffice as a short term solution.

AB#6059 is tracking the longer term solution.